proposals/lip-define_new_transaction_schema.md

```diff
@@ -0,0 +1,241 @@
+| **Configurable Constants** |          |**Mainchain Value**                |                                                                             |
+| `MAX_PARAMS_SIZE`          |uint32    | 14 KiB (14*1024 bytes)    |   The maximum allowed length of the transaction parameters.                         |
+
+
+### Type Definition
+
+| Name                | Type   | Validation                                       | Description                                                          |
+|---------------------|--------|--------------------------------------------------|----------------------------------------------------------------------|
+| `SignatureEd25519` | bytes   | Must be of length `ED25519_SIGNATURE_LENGTH`.  | Used for Ed25519 signatures.  |
+| `PrivateKeyEd25519` | bytes   | Must be of length `ED25519_PRIVATE_KEY_LENGTH`. | Used for Ed25519 private keys. |
+| `Transaction` | object | Must follow the `transactionSchema` schema with the only difference that `params` property is not serialized and contains the values of parameters of `paramsSchema` for the corresponding transaction. | An object representing a non-serialized transaction.                                        |
+
+### JSON Schema
+
+Transactions are serialized using `transactionSchema` given below.
+
+
+```java
+transactionSchema = {
+    "type": "object",
+    "required": [
+        "module",
+        "command",
+        "nonce",
+        "fee",
+        "senderPublicKey",
+        "params",
+        "signatures"
+    ],
+    "properties": {
+        "module": {
+            "dataType": "string",
+            "fieldNumber": 1
+        },
+        "command": {
+            "dataType": "string",
+            "fieldNumber": 2
+        },
+        "nonce": {
+            "dataType": "uint64",
+            "fieldNumber": 3
+        },
+        "fee": {
+            "dataType": "uint64",
+            "fieldNumber": 4
+        },
+        "senderPublicKey": {
+            "dataType": "bytes",
+            "length": ED25519_PUBLIC_KEY_LENGTH,
+            "fieldNumber": 5
+        },
+        "params": {
+            "dataType": "bytes",
+            "fieldNumber": 6
+        },
+        "signatures": {
+            "dataType": "array",
+            "items": {
+                "dataType": "bytes",
+                "length": ED25519_SIGNATURE_LENGTH 
+            },
+            "fieldNumber": 7
+        }
+    }
+}
+```
+
+
+#### Validation
+
+For a transaction `trs` to be valid, it must satisfy the following:
+
+* `trs` must follow the `transactionSchema`. 
+* `trs.params` is of length less than or equal to `MAX_PARAMS_SIZE` .
+
+The validity of each transaction is verified in the [static validation stage][lip-0055#block-processing] of the block processing. 
+
+
+### Serialization
+
+The serialization of an object of type `Transaction` is described in the following pseudocode.
+
+```python
+def encode(transactionSchema: LiskJSONSchema, trs: Transaction) -> bytes:
```

Comment:
Shouldn't `paramsSchema` also be an input parameter?

Reply:
Here I would rather obtain the paramsSchema from the transaction proeprties. I agree that `transactionSchema` is not needed (see this comment https://github.com/LiskHQ/lips/pull/155#discussion_r908507070)

Reply:
Can be resolved.